### PR TITLE
test: migrate from tap to node:test and c8

### PR DESCRIPTION
**Pull request by inyourtime**
#### Checklist
This PR migrates from `tap` to `node:test` and `c8`

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gurgunday — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -53,5 +53,11 @@
   },
   "publishConfig": {
     "access": "public"
+  },
+  "c8": {
+    "exclude": [
```
Comment (gurgunday) [inline package.json:58]: Is this necessary? I think removing this would still make the tests pass, no?

Comment (inyourtime) [reply package.json]: @gurgunday I removed it. The tests pass, but it looks like the c8 code coverage is also checking the examples folder. I'm not sure if the CI can pass.

File: test/example.test.js [hunk 2]
```diff
@@ -1,55 +1,56 @@
 'use strict'
 
-const t = require('tap')
-const test = t.test
+const { test } = require('node:test')
 const { rimrafSync } = require('rimraf')
-const build = require('./example')
+const build = require('../examples/example')
```
Comment (gurgunday) [inline test/example.test.js:5]: Well, we're still importing from those examples that's why

I don't know why they are mixed like this but if they are tests please leave them all in the tests folder, or completely separate tests and examples so we don't import them from the tests folder

**Review by gurgunday — APPROVED** [review]
lgtm

**Review by climba03003 — APPROVED** [review]

**Review by dancastillo — APPROVED** [review]
LGTM

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**